clang/lib/CConv/CConv.cpp

```diff
@@ -109,14 +109,15 @@ ArgumentsAdjuster getIgnoreCheckedPointerAdjuster() {
     bool HasAdjuster = false;
     for (size_t i = 0, e = Args.size(); i < e; ++i) {
       StringRef Arg = Args[i];
-      AdjustedArgs.push_back(Args[i]);
+      AdjustedArgs.push_back(Arg);
       if (Arg == "-fcheckedc-convert-tool") {
         HasAdjuster = true;
-        break;
       }
     }
-    if (!DisableCCTypeChecker && !HasAdjuster)
+    if (!DisableCCTypeChecker && !HasAdjuster) {
+      AdjustedArgs.push_back("-Xclang");
       AdjustedArgs.push_back("-fcheckedc-convert-tool");
```

Comment:
I'm a little confused about this, too -- it seems that `-fcheckedc-convert-tool` was already "pushed back" above; won't this add it twice?

Reply:
If it was already added, then the boolean HasAdjuster becomes true and this if statement doesn't get executed.

Reply:
Oh I see. But then it would seem you still need to add `-Xclang`? See comment below.

Reply:
Yeah I thought so too, but when I ran it by Aravind, he said not to add -Xclang if the user themselves specified the flag `fc to be conservative. In any case, I'll look up Xclang and see how it works. Like before, the errors David was getting were weird and not reproducible, so hopefully this flag is the trick.